### DISCUSSION TITLE
How to find all users having a permission ?

### DISCUSSION THREAD

**Discussion by FredVi** [casbin/casbin#1231] [Q&A] [answered]
Hi guys,

I'd want to use casbin to modelize my very simple RBAC policy.
The model looks like this :

```ini
[request_definition]
r = sub, perm

[policy_definition]
p = sub, perm

[role_definition]
g = _, _

[policy_effect]
e = some(where (p.eft == allow))

[matchers]
m = g(r.sub, p.sub) && r.perm == p.perm
```

Permissions can be assigned to roles. Roles can inherit other roles.
Permissions can be assigned to users and roles can be assigned to users... Quite straightforward.

And here is a policy example :

```csv
p, alice, read_object
p, bob, write_object
p, object_admin, read_object
p, object_admin, write_object
g, tom, object_admin
```

So I can easily get all permissions for user tom by using the GetImplicitPermissionsForUser method.
But, is there a way to get all users that have a given permission ?
Lets say I want to get all users that have the permission write_object (tom and bob); can I do this (in an efficient way) using casbin APIs ?
Thanks for your help !

**Comment by hsluoyz** [comment, marked as answer]
@FredVi the new GetImplicitUsersForPermission() API has be added, see here: https://github.com/casbin/casbin/issues/1233

**Reply by FredVi** [reply]
Ok, thanks.
